src/pages/WorkspacesList/index.tsx

```diff
@@ -45,51 +47,95 @@ export class WorkspacesList extends React.PureComponent<Props> {
     this.debounce = container.get(Debounce);
   }
 
-  // This method is called when the component is first added to the document
-  public componentDidMount(): void {
-    this.ensureDataFetched();
+  private buildWorkspaceRow(workspace: che.Workspace): React.ReactNodeArray {
+    const actions = this.buildActions(workspace);
+    return ([
+      <span
+        key={`${workspace.id}_1`}
+        onClick={() => this.onRowClick(workspace)}
+      >
+        <WorkspaceIndicator
+          key={`${workspace.id}_1_1`}
+          status={workspace.status}
+        />
+        {workspace.namespace}/{workspace.devfile.metadata.name}
+      </span>,
+      <span
+        key={`${workspace.id}_2`}
+        onClick={() => this.onRowClick(workspace)}
+      >
+        -
+      </span>,
+      <span
+        key={`${workspace.id}_3`}
+        onClick={() => this.onRowClick(workspace)}
+      >
+        {workspace.devfile.projects ? workspace.devfile.projects.length : '-'}
+      </span>,
+      <span
+        key={`${workspace.id}_4`}
+        onClick={() => this.onRowClick(workspace)}
+      >
+        {workspace.attributes && workspace.attributes.stackName ? workspace.attributes.stackName : ''}
+      </span>,
+      <span key={`${workspace.id}_5`}>
+        {actions}
+      </span>
+    ]);
   }
 
-  // This method is called when the route parameters change
-  public componentDidUpdate(): void {
-    this.ensureDataFetched();
+  private buildActions(workspace: che.Workspace): React.ReactNode {
```

Comment:
You are using ```WorkspaceStatus[workspace.status]``` more than 5 times. Let's create a constant.
For example
```
const status = WorkspaceStatus[workspace.status];
if (status === WorkspaceStatus.STARTING || status === WorkspaceStatus.RUNNING) {
...
```

Reply:
makes sense, I'll provide a fixup